projects/jsf/src/lib/models/integer-data-item.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { SchemaIntegerOptions } from '../form-data-item.service';
+import { FormDataItemType } from './form-data-item';
+import { IntegerDataItem } from './integer-data-item';
+
+describe('IntegerDataItem', () => {
+  const key = 'key';
+  const label = 'label';
+  const tooltip = 'tooltip';
+  const helpText = 'helpText';
+  const pathParts = ['parentKey', key];
+  const type = FormDataItemType.Integer;
+  const value = undefined;
+  const placeholder = 'placeholder';
+  const isReadOnly = false;
+  const isHidden = false;
+
+  let schemaIntegerOptions: SchemaIntegerOptions;
+
+  beforeEach(() => {
+    schemaIntegerOptions = {} as SchemaIntegerOptions;
+  });
+
+  it('should set placeholder', () => {
+    schemaIntegerOptions.placeholder = placeholder;
+    const result = new IntegerDataItem(key, label, tooltip, helpText, true, pathParts, type, value, isReadOnly, isHidden, [], schemaIntegerOptions);
+    expect(result.placeholder).toEqual(placeholder);
+  });
+
+  it('should set the maximum and minimum lengths', () => {
+    const min = 5;
+    const max = 10;
+    schemaIntegerOptions.minLength = min;
+    schemaIntegerOptions.maxLength = max;
+    const result = new IntegerDataItem(key, label, tooltip, helpText, true, pathParts, type, value, isReadOnly, isHidden, [], schemaIntegerOptions);
+
+    expect(result.validationSettings.length.minLength).toEqual(min);
+    expect(result.validationSettings.length.maxLength).toEqual(max);
+  });
+
+  it('should not set the min and max lengths if they are not present', () => {
+    const result = new IntegerDataItem(key, label, tooltip, helpText, true, pathParts, type, value, isReadOnly, isHidden, [], schemaIntegerOptions);
+    expect(result.validationSettings.length.minLength).toBeUndefined();
+    expect(result.validationSettings.length.maxLength).toBeUndefined();
+  });
```

Comment:
Do we need length validation on numbers? I didn't see this in the PR description and I couldn't find this in the json schema docs for an integer.

Reply:
When I broke integer out of the string class, these keywords were not excluded for integer fields. I didn't want to break anything, but you're right, they are not part of the JSON schema definition for integers